audformat/core/utils.py

```diff
@@ -1622,9 +1625,29 @@ def job(file: str) -> pd.Timedelta:
                 progress_bar=verbose,
                 task_description='Read duration',
             )
-            ends.values[idx_nat] = durs
 
-            index = segmented_index(files, starts, ends)
+            # Now replace all NaT entries in end
+            # by the collected duration values
+
+            # Reverse durs list as pop()
+            # will return last enrty first
+            durs.reverse()
+
+            # Add an additional entry to the list
+            # as the very first entry
+            # passed by map() is the whole index
+            # which will also trigger a call
+            # to durs.pop()
+            # if ends contains only a single entry
+            if len(ends) == 1:
+                durs.append(0)
```

Comment:
This one is spooky and I don't know why it is needed, it's just empirical result.

In general I'm not very happy that we use `durs.pop()`, but I didn't found a different solution. The problem is that we somehow first collect all durations and then need to update the entries using `map()`. If `map()` would not only see the `end` values, but also the `file` values we could use a dictionary for the mapping to the duration values, but I'm not aware of how to make the `file` or the current index (like enumerate) visible inside `map()`.